src/lib/mongoBackend/dateExpiration.cpp

```diff
@@ -0,0 +1,200 @@
+*
+* Get the ISO8601 Expiration Date information for the given
+* ContextAttribute provided as parameter.
+*
+* It returns true, except in the case of error (in which in addition errDetail gets filled)
+*/
+static bool getDateExpiration
+(
+  const ContextAttribute*  caP,
+  mongo::Date_t*           dateExpiration,
+  std::string*             errDetail
+)
+{
+  if ((caP->type == DATE_TYPE) || (caP->type == DATE_TYPE_ALT))
+  {
+    *dateExpiration = mongo::Date_t(caP->numberValue*1000);
+
+    return true;
+  }
+
+  LM_E(("Runtime Error (attribute detected as date expiration but invalid type: %s)", caP->type.c_str()));
+  *errDetail = "error processing date expiration attribute, see log traces";
+
+  return false;
+}
+
+
+
+/* ****************************************************************************
+*
+* processDateExpirationAtEntityCreation -
+*
+* This function process the context attribute vector, searching for an attribute meaning
+* date expiration for the entity. In that case, it fills dateExpiration pointer. If a date expiration attribute is not found, then
+* dateExpiration is an empty pointer.
+*
+* This function always return true (no matter if the attribute was found or not), except in an
+* error situation, in which case errorDetail is filled.
+*/
+bool processDateExpirationAtEntityCreation
+(
+  const ContextAttributeVector&  caV,
+  mongo::Date_t*                 dateExpiration,
+  std::string*                   errDetail,
+  OrionError*                    oe
+)
+{
+  for (unsigned ix = 0; ix < caV.size(); ++ix)
+  {
+    const ContextAttribute* caP = caV[ix];
+
+    if (caP->name != ENT_EXPIRATION)
+    {
+      continue;
+    }
+
+    if (!getDateExpiration(caP, dateExpiration, errDetail))
+    {
+      oe->fill(SccBadRequest, *errDetail, "BadRequest");
+      return false;
+    }
+  }
+
+  return true;
+}
+
+
+
+/* ****************************************************************************
+*
+* processDateExpirationAtUpdateAttribute -
+*/
+bool processDateExpirationAtUpdateAttribute
+(
+  const ContextAttribute*  targetAttr,
+  mongo::Date_t*           dateExpiration,
+  bool*                    replaceDate,
+  std::string*             errDetail,
+  OrionError*              oe
+)
+{
+
+  /*
+   * If the name of the target attribute is the date expiration
+   * check for a number value to be used in the mongo::Date_t constructor.
+   * If it is an empty value, is interpreted as a date expiration deletion and a NULL pointer is set
+   * If valid value, the replaceDate boolean is set to true, in order to manage the new date value
+    * in case of replace operation
+   */
+
+  if (targetAttr->name == ENT_EXPIRATION)
+  {
+    if (targetAttr->numberValue)
+    {
+	  if (!getDateExpiration(targetAttr, dateExpiration, errDetail))
+	  {
+	    oe->fill(SccBadRequest, *errDetail, "BadRequest");
+	    return false;
+	  }
+	  else
+	  {
+        *replaceDate = true;
+	  }
+    }
+    else
+    {
+      *dateExpiration = 0;
```

Comment:
It should be the same, as reported in
https://stackoverflow.com/questions/176989/do-you-use-null-or-0-zero-for-pointers-in-c
In any case, I will check if there are any difference

Reply:
Ok.

However, if you at the end needs to null-ify a point I think the right one would be:

```
dateExpiration = 0;
```

As far as I understand the `*dateExpiration = 0;` doesn't set the point to NULL, but updates the contain of the variable pointed by the pointer to 0.

Having said that, if you need it, use `dateExpiration = NULL` instead of `dateExpiration = 0`. Both are equivalente, but from a point of view of Orion code style, we typically use NULL.

Reply:
Sure, this is clear :) my intention actually was to set the value to 0 not the pointer. By setting 0 as dateExpiration value, mongo::Date_t is considered "false". As said previously, I'm reviewing all the logic of setting pointers and values for dateExpiration object in these utilities functions.

Reply:
Fixed in 5a4703d